### PR TITLE
nix-prefetch-git: shellcheck fixes

### PR DESCRIPTION
**Pull request by zimbatm**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Used [shellcheck](https://github.com/koalaman/shellcheck) to validate
the script and fixed any resulting escaping and ambiguity issues.

Was originally bundled with #14030

### CODE REVIEW

**Comment by nbp** [comment]
I guess, you only need to make shellcheck run as part of the release.nix process, and we should be good to get this as a permanent linter.

**Comment by zimbatm** [comment]
Do you mean add a new kind of job in pkgs/top-level/release.nix ?

**Comment by nbp** [comment]
Yes, that's what I mean.

File: pkgs/build-support/fetchgit/nix-prefetch-git [hunk 1]
```diff
@@ -64,7 +68,7 @@ for arg; do
             --builder) builder=true;;
             --help) usage; exit;;
             *)
-                argi=$(($argi + 1))
+                argi=$((argi + 1))
```
Comment (benley) [inline pkgs/build-support/fetchgit/nix-prefetch-git]: since this is explicitly using bash you could replace this line with `((argi++))`

**Comment by domenkozar** [comment]
Let's fix this :)

File: pkgs/build-support/fetchgit/nix-prefetch-git (hunk 1 above)
Comment (zimbatm) [inline pkgs/build-support/fetchgit/nix-prefetch-git]: b6d4516ecbf7a91887caad22ae2a627cc0e8cb93

**Comment by zimbatm** [comment]
@domenkozar fixed !

**Comment by benley** [comment]
Woot!

**Pull request merged**